kernel/src/test/scala/me/rexim/morganey/ChurchPairConverterSpec.scala

```diff
@@ -17,32 +17,28 @@ class ChurchPairConverterSpec extends FlatSpec with Matchers with TestTerms {
     decodePair(pair) should be (Some((lvar("x"), lvar("y"))))
   }
 
-  "An identity function" should "be list converted to a list with itself" in {
-    decodeList(I(x)) should be (List(I(x)))
+  "An identity function" should "not be recognized as a list" in {
+    decodeList(I(x)) should be (None)
   }
 
   "A church list" should "be converted to some list" in {
-    val rawTerm = "(λ z . ((z x) (λ z . ((z x) y))))"
-    val term = LambdaParser.parse(LambdaParser.term, rawTerm).get
-    val expectedList = "xxy".map(c => lvar(c.toString)).toList
-
-    decodeList(term) should be (expectedList)
+    val inputTerm = pair(x, pair(x, pair(y, zero)))
+    val expectedList = Some("xxy".map(c => lvar(c.toString)).toList)
+    decodeList(inputTerm) should be (expectedList)
   }
 
   "A church list of number" should "be converted to some list of numbers" in {
```

Comment:
Also I'm not sure but I think you should capitalize Church. [Wikipedia](https://en.wikipedia.org/wiki/Church_encoding) does.